### PR TITLE
Fix(CX-1496) enable screen rotation on android tablets

### PR DESCRIPTION
**Pull request by lordkiz**
The type of this PR is: **Enhancement**

This PR resolves [CX-1496]

#### Phone
https://user-images.githubusercontent.com/18648835/123669171-1a8cbb00-d83c-11eb-98de-713a06de3ae2.mov

#### Tablet

https://user-images.githubusercontent.com/18648835/123669354-44de7880-d83c-11eb-8d16-8dec1b58fcf4.mov

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a [CHANGELOG.yml](/CHANGELOG.yml) entry or my changes do not require one.

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

- Allow screen rotations on android tablets

#### Dev changes

-

[CX-1496]: https://artsyproduct.atlassian.net/browse/CX-1496

### CODE REVIEW

File: patches/react-native-screens+2.17.1.patch [hunk 1]
```diff
@@ -14,3 +14,16 @@ index b05fce8..b94d30c 100644
...
+@@ -96,6 +99,12 @@ public class ScreenStackHeaderConfig extends ViewGroup {
+       mToolbar.setBackgroundColor(tv.data);
+     }
+     mToolbar.setClipChildren(false);
++
++    // ARTSY FIX: If tablet, instantiate with the default screen orientation
++    boolean isTab = context.getResources().getConfiguration().smallestScreenWidthDp >= 600;
++    if (isTab) {
++      mScreenOrientation = ActivityInfo.SCREEN_ORIENTATION_UNSPECIFIED;
++    }
```
Comment (MounirDhahri) [inline patches/react-native-screens+2.17.1.patch:26]: **issue:** I think we might be better using `isTablet` as defined here https://github.com/artsy/eigen/blob/99a6985bd8c11fefd5b56d74e61a612fa2157c5a/android/app/src/main/java/net/artsy/app/ArtsyNativeModule.java#L157-L161
This is a popular implementation for `isTablet` that's inspired by the way google did it in one of its Google I/O apps

**Review by MounirDhahri — COMMENTED** [review]
This looks good to me, I left a minor comment about identifying tablets. Also, did you tried upgrading the react-native-screens package? I think that might also fix the issue. See [react-native-screens patch in Hacks.md](https://github.com/artsy/eigen/blob/a3a0a412d5ea724067cdcaf20a48fd511dfd748d/HACKS.md#L189)

File: patches/react-native-screens+2.17.1.patch (hunk 1 above)
Comment (lordkiz) [reply patches/react-native-screens+2.17.1.patch:26]: Thanks @MounirDhahri . Upgrading to 3.4.0 seems to have fixed the issue and we no longer need the patch. I will try a few nav related stuffs to ensure this version plays nice with our react-navigation version on both platforms before updating this PR.

Comment (MounirDhahri) [reply patches/react-native-screens+2.17.1.patch]: This is great to hear, thank you 🙌

**Comment by lordkiz** [comment]
PR updated!

**Comment by MounirDhahri** [comment]
#squashongreen

**Pull request merged**